seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -1074,6 +1091,72 @@ public ConnectorPackageService getConnectorPackageService() {
         return connectorPackageService;
     }
 
+    public PendingJobsResponse getPendingJobs(Map<String, String> tags, Long jobId, int limit) {
+        Collection<PendingJobInfo> allPendingJobs =
+                new ArrayList<>(pendingJobQueue.getJobIdMap().values());
+
+        List<PendingJobInfo> selectedJobs = new ArrayList<>();
+        if (jobId != null) {
+            PendingJobInfo pendingJobInfo = pendingJobQueue.getById(jobId);
+            if (pendingJobInfo != null) {
+                selectedJobs.add(pendingJobInfo);
+            }
+        } else {
+            selectedJobs.addAll(allPendingJobs);
+            selectedJobs.sort(Comparator.comparingLong(PendingJobInfo::getEnqueueTimestamp));
+            if (limit > 0 && selectedJobs.size() > limit) {
+                selectedJobs = new ArrayList<>(selectedJobs.subList(0, limit));
+            }
+        }
+
+        ResourceManager resourceManager = getResourceManager();
+        List<PendingJobDiagnostic> diagnostics = new ArrayList<>();
+        for (PendingJobInfo jobInfo : selectedJobs) {
+            PendingJobDiagnostic diagnostic = jobInfo.getLastSnapshot();
+            if (diagnostic == null) {
+                diagnostic =
+                        PendingDiagnosticsCollector.collectJobDiagnostic(
+                                jobInfo, tags, resourceManager);
+                if (diagnostic != null) {
+                    diagnostic.setCheckCount(jobInfo.getCheckTimes());
+                }
+            }
+            if (diagnostic != null) {
+                diagnostics.add(diagnostic);
+            }
+        }
+
+        PendingJobsResponse response = new PendingJobsResponse();
+        response.setPendingJobs(diagnostics);
+        response.setClusterSnapshot(
+                PendingDiagnosticsCollector.collectClusterSnapshot(resourceManager, tags));
+        response.setQueueSummary(buildQueueSummary(allPendingJobs, diagnostics));
```

Comment:
```suggestion
        response.setQueueSummary(buildQueueSummary(selectedJobs, diagnostics));
```
Depending on the limit, queueSummary may not match the diagnostics returned.